### PR TITLE
Remove old data migrations

### PR DESCRIPTION
**Pull request by despo**
Maintenance task as there is no reason to keep old data migrations.

### CODE REVIEW

File: lib/tasks/data.rake [hunk 1]
```diff
@@ -1,41 +1,5 @@
...
-  'DataMigrations::BackfillValidationErrorsServiceColumn',
-  'DataMigrations::BackfillSelectedBoolean',
-  'DataMigrations::DeleteAllSiteAudits',
-  'DataMigrations::RemoveIncorrectHesaCodes',
-  'DataMigrations::SpecifyExportTypeForTADExports',
-  'DataMigrations::SpecifyExportTypeForNotificationExports',
-  'DataMigrations::FixMisspellingOfCaribbeanEthnicGroupAndSetHesaCodes',
-  'DataMigrations::TrimQualificationDegreeTypes',
-  'DataMigrations::BackfillExportType',
-  'DataMigrations::FixLatLongFlipFlops',
```
Comment (cwrw) [inline lib/tasks/data.rake:37]: I think this migration might not have been run. Can we check which ones have been run on prod and remove those but keep the rest?

Comment (despo) [reply lib/tasks/data.rake:37]: Updated. Should all of these be left? And do you know why some have never been executed.

Comment (duncanjbrown) [reply lib/tasks/data.rake:37]: this one in particular we never finished due to issues with timeouts, I think

File: lib/tasks/data.rake [hunk 2]
```diff
@@ -1,41 +1,9 @@
...
-  'DataMigrations::PopulateApplicationChoiceCurrentRecruitmentCycleYear',
-  'DataMigrations::PopulateApplicationChoiceProviderIds',
   'DataMigrations::RemoveDuplicateProvider',
-  'DataMigrations::BackfillReferencesCompleted',
-  'DataMigrations::CleanseEocChasersSentData',
-  'DataMigrations::BackfillSetupInterviewsPermission',
-  'DataMigrations::FixEmptyOfferConditions',
-  'DataMigrations::RemoveSurplusReferenceSelections',
-  'DataMigrations::BackfillCandidateAPIUpdatedAt',
   'DataMigrations::BackfillNoneHesaDisabilitiesCodes',
```
Comment (cwrw) [inline lib/tasks/data.rake:4]: Could we remove `BackfillNoneHesaDisabilitiesCodes` and `DeleteUuidlessCourses`, and keep the last two as those have def been run?

**Review by cwrw — APPROVED** [review]
Just two deletions but otherwise 👍

**Pull request merged**